src/core_modules/capture-core/components/Pages/New/NewPage.container.js

```diff
@@ -80,6 +81,10 @@ export const NewPage: ComponentType<{||}> = () => {
     };
 
     const writeAccess = useUserWriteAccess(currentScopeId);
+
+    const isUserInteractionInProgress: boolean = useSelector(
+        state => dataEntryHasChanges(state, 'singleEvent-newEvent') || dataEntryHasChanges(state, 'newPageDataEntryId-newEnrollment'));
```

Comment:
Couldn't this lead to a potential problem if we move between the different forms (for example: singleEvent has changes, but we're in the newEnrollment form without changes?)  Can we be more specific here on what state we're checking?

Reply:
Hm about this, this is the reason I am trying to clean up the store from all the relevant information of a data entry when the user is leaving the new registration screen. I thought this is anyway a good practise (cleaning up the form when you leave the page).

So if the user is registering a single event and then they leave the page all the relevant data will be cleaned. Then for instance when they try to register a new person they will have a "clean" redux store in regards to forms and data entries. 

You see any wholes in my thoughts in this regard?

Reply:
I agree cleaning up on exit is good practice 👍 Did some testing now after your update, and the clean up happens when you deselect the org.unit or the program, but if you click `Cancel` or `Submit` in the form it doesn't. Could we do the the clean up on unmount of the most applicable component? (note: make sure the clean up doesn't trigger when we navigate to the relationship part through add relationship to an event). Might be edge cases that we need to handle.

Reply:
We chatted about this on a meeting. Now it should be fixed